### PR TITLE
Fix README

### PR DESCRIPTION
**Pull request by MusiKid**
This commit fixes the table of content and introduce some reordering for the titles.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -259,7 +275,7 @@ cargo test
 
 ## Minimum requirements
 
-- GCC 4.9 or higher is required (if parsing coverage artifacts generated by GCC).
+* GCC 4.9 or higher is required (if parsing coverage artifacts generated by GCC).
```
Comment (marco-c) [inline README.md:278]: Why this change?

**Comment by marco-c** [comment]
Thanks!

File: README.md (hunk 1 above)
Comment (MusiKid) [reply README.md:278]: It was just to use the same character everywhere, this isn't really necessary.

Comment (marco-c) [reply README.md:278]: OK, WFM

**Review by marco-c — APPROVED** [review]
